### PR TITLE
Consolidate spawnProc logic

### PR DESCRIPTION
**Pull request by kevinburke**
Use separate build files only for the parts of these functions that
differ, which allows us to consolidate the logic and reduces the
chance of errors.

### CODE REVIEW

File: proc.go [hunk 1]
```diff
@@ -2,11 +2,53 @@ package main
...
 	"sync"
 	"time"
 )
 
+// spawnProc starts the specified proc, and returns any error from running it.
+func spawnProc(proc string, errCh chan<- error) {
+	procObj := procs[proc]
+	logger := createLogger(proc, procObj.colorIndex)
+
+	cs := append(cmdStart, procObj.cmdline)
```
Comment (mattn) [inline proc.go]: I'm not sure but `append(cmdStart, ...)` modify cmdStart. So it may break second call of spawnProc.

Comment (kevinburke) [reply proc.go]: I don't think so... see https://play.golang.org/p/iIez3WbTbR9

```go
package main

import "fmt"

func main() {
	a := []string{"one", "two"}
	c := append(a, "three")
	d := append(a, "four")
	fmt.Printf("%#v\n", a)
	fmt.Printf("%#v\n", c)
	fmt.Printf("%#v\n", d)
}
```

```
[]string{"one", "two"}
[]string{"one", "two", "three"}
[]string{"one", "two", "four"}
```

Comment (mattn) [reply proc.go]: Ah, sorry. cap of literal is same as length.

**Comment by mattn** [comment]
Thank you

**Pull request merged**